src/csharp/Extensions/Microsoft.Spark.Extensions.Hyperspace/Index/Builder.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.Extensions.Hyperspace.Index
+{
+    /// <summary>
+    /// Builder for <see cref="IndexConfig"/>.
+    /// </summary>
+    public sealed class Builder : IJvmObjectReferenceProvider
+    {
+        private readonly JvmObjectReference _jvmObject;
+
+        internal Builder(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Updates index name for <see cref="IndexConfig"/>.
+        /// </summary>
+        /// <param name="indexName">Index name for the <see cref="IndexConfig"/>.</param>
+        /// <returns>An <see cref="Builder"/> object with updated indexname.
+        /// </returns>
+        public Builder IndexName(string indexName) =>
+            new Builder((JvmObjectReference)_jvmObject.Invoke("indexName", indexName));
+
+        /// <summary>
+        /// Updates column names for <see cref="IndexConfig"/>.
+        ///
+        /// Note: API signature supports passing one or more argument.
+        /// </summary>
+        /// <param name="indexedColumn">Indexed column for the
+        /// <see cref="IndexConfig"/>.</param>
+        /// <param name="indexedColumns">Indexed columns for the
+        /// <see cref="IndexConfig"/>.</param>
+        /// <returns>An <see cref="Builder"/> object with updated indexed
+        /// columns.</returns>
+        public Builder IndexBy(string indexedColumn, params string[] indexedColumns) =>
+            new Builder(
```

Comment:
Can we return the same builder instance? Check `SparkSession.Builder`.

Reply:
Changed to return same Builder.